### ISSUE TITLE
Search bar is not independent of Replace bar

### ISSUE THREAD

**Issue by aramazano** [tsujan/FeatherPad#123]
[Replace] automatically activates Search (Find) bar, but doesn't automatically inactivate it. Since replace bar has its own Find area, there is no need to auto-activate the find bar along with replace.

Currently Find/Replace/Jump buttons work like this:

[Jump] -> Shows jump bar -> [Jump] -> Hides jump bar (as expected)

[Find] -> Shows search bar -> [Find] -> Hides search bar (as expected)

[Replace] -> Shows both search and replace bars (unexpected - should only show replace bar)

-----

Upon pressing [Replace] (with both search and replace bars being shown):

[Find] -> Focus to search bar (should hide search bar instead) -> [Find] -> Hide both search and replace bars (unexpected - should hide only the search bar)

[Replace] -> Hide only replace bar (as expected - now we have only search bar) -> [Find] -> Focus to search bar (should hide find bar) -> [Find] -> Hide search bar

To sum it up, [Jump] button correctly toggles only the jump bar. But [Replace] and [Find] buttons intertwine with one another.

I would like to suggest one of two ways:

1) Getting rid of "Find" line in the replace bar, and utilizing the search bar for the "find" sub-function of replace. This way, replace function is dependent on the search bar and thus activating replace should also activate search bar automatically, which is how it behaves now. As long as the replace bar is shown, search bar cannot be hidden by itself (probably "search" button in the toolbar would be disabled)

2) Leaving everything as they are now, except making search and replace bars totally independent of each other (just like the jump bar).

**Comment by tsujan** [comment]
> [Replace] automatically activates Search (Find) bar, but doesn't automatically inactivate it. Since replace bar has its own Find area, there is no need to auto-activate the find bar along with replace.

That's not only intentional but needed because Replace uses the `Whole Word` and `Match Case` buttons.

**Comment by tsujan** [comment]
> Getting rid of "Find" line in the replace bar, and utilizing the search bar for the "find" 

Never! I hate that in some text editors.

**Comment by tsujan** [comment]
And, more importantly, using search *alongside* replacing is something I won't remove from FeatherPad -- I use that a lot.

**Comment by aramazano** [comment]
@tsujan I understand your point. Activating both bars is not really a problem. My main concern is about the inactivation procedure. When I am working with the search bar already active, then toggling Replace behaves as expected. But when neither search bar nor replace bar is active (which happens the way I mostly work), then pressing [Replace] twice leaves a residual search bar behind, and I have to get out of my way to close it. This is an inconvenience for me, however minor.

OTOH, I can't assess whether this inconvenience justifies the extra bloat to conditionally close either both bars together (upon pressing Replace button), or just closing the replace bar and leaving the search bar intact, depending on whether the search bar was already active when replace bar launched.

There might be a different, simpler, more streamlined workflow for the overall desired behavior but I can't think of one right now.

**Comment by tsujan** [comment]
Closing the search-bar on closing the replacement dock would be really annoying and I would consider it a usability issue in any text editor.
